platform/ios/src/MGLMapAccessibilityElement.mm

```diff
@@ -52,16 +52,17 @@ - (instancetype)initWithAccessibilityContainer:(id)container feature:(id<MGLFeat
         NSString *nameAttribute = [NSString stringWithFormat:@"name_%@", languageCode];
         NSString *name = [feature attributeForKey:nameAttribute];
 
+        NSString *dominantScript;
         if (name == nil && [feature attributeForKey:@"name"] != nil) {
             name = [feature attributeForKey:@"name"];
-            languageCode = [feature attributeForKey:@"name_script"];
+            dominantScript = [feature attributeForKey:@"name_script"];
         }
 
         // If a feature hasn’t been translated into the preferred language, it
         // may be in the local language, which may be written in another script.
         // Attempt to transform to the script of the preferred language, keeping
         // the original string if no transform exists or if transformation fails.
-        NSString *dominantScript = [NSOrthography mgl_dominantScriptForMapboxStreetsLanguage:languageCode];
+        dominantScript = [NSOrthography mgl_dominantScriptForMapboxStreetsLanguage:languageCode];
```

Comment:
This avoids overriding the line above that sets `dominantScript` based on `name_script`:

```suggestion
        if (!dominantScript) {
            dominantScript = [NSOrthography mgl_dominantScriptForMapboxStreetsLanguage:languageCode];
        }
```